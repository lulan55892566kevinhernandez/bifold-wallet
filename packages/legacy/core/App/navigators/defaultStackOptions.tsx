import React from 'react'

import HeaderTitle from '../components/texts/HeaderTitle'
import { Theme } from '../theme'

export function createDefaultStackOptions({ ColorPallet }: Theme) {
  return {
    headerTintColor: ColorPallet.grayscale.white,
    headerShown: true,
    headerBackTitleVisible: false,
    headerStyle: {
      elevation: 0,
      shadowOffset: { width: 0, height: 6 },
      shadowRadius: 6,
      shadowColor: ColorPallet.grayscale.black,
      shadowOpacity: 0.15,
      borderBottomWidth: 0,
    },
    headerTitleAlign: 'center' as 'center' | 'left',
    headerTitle: (props: { children: React.ReactNode }) => <HeaderTitle {...props} />,
  }
}
