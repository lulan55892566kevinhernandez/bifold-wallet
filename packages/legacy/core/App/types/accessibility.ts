import { AccessibilityProps } from 'react-native'

export type ImportantForAccessibility = AccessibilityProps['importantForAccessibility']
