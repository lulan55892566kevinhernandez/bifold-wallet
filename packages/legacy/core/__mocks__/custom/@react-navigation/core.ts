const navigate = jest.fn()
const dispatch = jest.fn()
const replace = jest.fn()
const navigation = {
  __timestamp: process.hrtime(),
  navigate,
  replace,
  setOptions: jest.fn(),
  getParent: () => {return {
    navigate,
    dispatch,
    replace,
  }},
  getState: jest.fn(() => ({
    index: jest.fn(),
  })),
  goBack: jest.fn(),
  pop: jest.fn(),
  reset: jest.fn(),
  isFocused: () => true,
  dispatch,
}

const useNavigation = () => {
  return navigation
}

const useIsFocused = () => {
  return true
}

export { useNavigation, useIsFocused }
