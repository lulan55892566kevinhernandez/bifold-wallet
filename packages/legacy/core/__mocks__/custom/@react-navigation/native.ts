const useFocusEffect = jest.fn()
const createNavigatorFactory = jest.fn()

export { useFocusEffect, createNavigatorFactory }
