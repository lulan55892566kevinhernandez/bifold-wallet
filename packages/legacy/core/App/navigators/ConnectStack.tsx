import { createStackNavigator } from '@react-navigation/stack'
import React from 'react'
import { useTranslation } from 'react-i18next'

import { useConfiguration } from '../contexts/configuration'
import { useStore } from '../contexts/store'
import { useTheme } from '../contexts/theme'
import NameWallet from '../screens/NameWallet'
import { ConnectStackParams, Screens } from '../types/navigators'
import { testIdWithKey } from '../utils/testable'

import { createDefaultStackOptions } from './defaultStackOptions'

const ConnectStack: React.FC = () => {
  const Stack = createStackNavigator<ConnectStackParams>()
  const theme = useTheme()
  const defaultStackOptions = createDefaultStackOptions(theme)
  const { scan } = useConfiguration()
  const { t } = useTranslation()
  const [store] = useStore()
  return (
    <Stack.Navigator
      // below is part of the temporary gating of the new scan screen tabs feature
      screenOptions={{ ...defaultStackOptions, headerShown: store.preferences.useConnectionInviterCapability }}
    >
      <Stack.Screen name={Screens.Scan} component={scan} />
      <Stack.Screen
        name={Screens.NameWallet}
        component={NameWallet}
        options={{ title: t('Screens.NameWallet'), headerBackTestID: testIdWithKey('Back') }}
      />
    </Stack.Navigator>
  )
}

export default ConnectStack
