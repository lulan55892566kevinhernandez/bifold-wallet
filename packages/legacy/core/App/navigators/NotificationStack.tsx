import { CardStyleInterpolators, createStackNavigator } from '@react-navigation/stack'
import React from 'react'
import { useTranslation } from 'react-i18next'

import HeaderRightHome from '../components/buttons/HeaderRightHome'
import { useConfiguration } from '../contexts/configuration'
import { useTheme } from '../contexts/theme'
import CommonDecline from '../screens/CommonDecline'
import CredentialDetails from '../screens/CredentialDetails'
import CredentialOffer from '../screens/CredentialOffer'
import ProofRequest from '../screens/ProofRequest'
import { NotificationStackParams, Screens } from '../types/navigators'

import { createDefaultStackOptions } from './defaultStackOptions'

const NotificationStack: React.FC = () => {
  const Stack = createStackNavigator<NotificationStackParams>()
  const theme = useTheme()
  const { t } = useTranslation()
  const defaultStackOptions = createDefaultStackOptions(theme)
  const { customNotification } = useConfiguration()

  return (
    <Stack.Navigator screenOptions={{ ...defaultStackOptions }}>
      <Stack.Screen
        name={Screens.CredentialDetails}
        component={CredentialDetails}
        options={{ title: t('Screens.CredentialDetails') }}
      />
      <Stack.Screen
        name={Screens.CredentialOffer}
        component={CredentialOffer}
        options={{ title: t('Screens.CredentialOffer') }}
      />
      <Stack.Screen
        name={Screens.ProofRequest}
        component={ProofRequest}
        options={{ title: t('Screens.ProofRequest') }}
      />
      <Stack.Screen
        name={Screens.CustomNotification}
        component={customNotification.component}
        options={{ title: t(customNotification.pageTitle as any) }}
      />
      <Stack.Screen
        name={Screens.CommonDecline}
        component={CommonDecline}
        options={() => ({
          title: t('Screens.Decline'),
          cardStyleInterpolator: CardStyleInterpolators.forVerticalIOS,
          headerShown: true,
          presentation: 'modal',
          headerLeft: () => null,
          headerRight: () => <HeaderRightHome />,
        })}
      />
    </Stack.Navigator>
  )
}

export default NotificationStack
