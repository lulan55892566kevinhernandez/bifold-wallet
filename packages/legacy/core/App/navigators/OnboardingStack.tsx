/* eslint-disable @typescript-eslint/no-non-null-assertion */
import { ParamListBase, RouteConfig, StackNavigationState, useNavigation } from '@react-navigation/core'
import {
  StackNavigationOptions,
  StackNavigationProp,
  createStackNavigator,
  StackScreenProps,
} from '@react-navigation/stack'
import { StackNavigationEventMap } from '@react-navigation/stack/lib/typescript/src/types'
import React from 'react'
import { useTranslation } from 'react-i18next'

import { TOKENS, useContainer } from '../container-api'
import { useConfiguration } from '../contexts/configuration'
import { DispatchAction } from '../contexts/reducers/store'
import { useStore } from '../contexts/store'
import { useTheme } from '../contexts/theme'
import NameWallet from '../screens/NameWallet'
import { createCarouselStyle } from '../screens/OnboardingPages'
import PINCreate from '../screens/PINCreate'
import PushNotification from '../screens/PushNotification'
import { AuthenticateStackParams, Screens } from '../types/navigators'

import { createDefaultStackOptions } from './defaultStackOptions'

interface CreatePINScreenParams extends StackScreenProps<ParamListBase, Screens.CreatePIN> {}

type ScreenOptions = RouteConfig<
  ParamListBase,
  Screens,
  StackNavigationState<ParamListBase>,
  StackNavigationOptions,
  StackNavigationEventMap
>

const OnboardingStack: React.FC = () => {
  const [, dispatch] = useStore()
  const { t } = useTranslation()
  const container = useContainer()
  const Stack = createStackNavigator()
  const theme = useTheme()
  const OnboardingTheme = theme.OnboardingTheme
  const carousel = createCarouselStyle(OnboardingTheme)
  const Onboarding = container.resolve(TOKENS.SCREEN_ONBOARDING)
  const { pages, splash, useBiometry } = useConfiguration()
  const defaultStackOptions = createDefaultStackOptions(theme)
  const navigation = useNavigation<StackNavigationProp<AuthenticateStackParams>>()
  const onTutorialCompleted = container.resolve(TOKENS.FN_ONBOARDING_DONE)(dispatch, navigation)
  const { screen: Terms } = container.resolve(TOKENS.SCREEN_TERMS)
  const Developer = container.resolve(TOKENS.SCREEN_DEVELOPER)
  const ScreenOptionsDictionary = container.resolve(TOKENS.OBJECT_ONBOARDINGCONFIG)
  const Preface = container.resolve(TOKENS.SCREEN_PREFACE)

  const onAuthenticated = (status: boolean): void => {
    if (!status) {
      return
    }

    dispatch({
      type: DispatchAction.DID_AUTHENTICATE,
    })
  }

  const OnBoardingScreen: React.FC = () => {
    return (
      <Onboarding
        nextButtonText={t('Global.Next')}
        previousButtonText={t('Global.Back')}
        disableSkip={true}
        pages={pages(onTutorialCompleted, OnboardingTheme)}
        style={carousel}
      />
    )
  }

  const CreatePINScreen: React.FC<CreatePINScreenParams> = (props) => {
    return <PINCreate setAuthenticated={onAuthenticated} {...props} />
  }

  const screens: ScreenOptions[] = [
    {
      name: Screens.Preface,
      component: Preface,
      options: () => {
        return {
          ...ScreenOptionsDictionary[Screens.Preface],
          title: t('Screens.Preface'),
        }
      },
    },
    {
      name: Screens.Splash,
      component: splash,
      options: ScreenOptionsDictionary[Screens.Splash],
    },
    {
      name: Screens.Onboarding,
      component: OnBoardingScreen,
      options: () => {
        return {
          ...ScreenOptionsDictionary[Screens.Onboarding],
          title: t('Screens.Onboarding'),
        }
      },
    },
    {
      name: Screens.Terms,
      options: () => ({
        ...ScreenOptionsDictionary[Screens.Terms],
        title: t('Screens.Terms'),
      }),
      component: Terms,
    },
    {
      name: Screens.CreatePIN,
      component: CreatePINScreen,
      initialParams: {},
      options: () => ({
        ...ScreenOptionsDictionary[Screens.CreatePIN],
        title: t('Screens.CreatePIN'),
      }),
    },
    {
      name: Screens.NameWallet,
      options: () => ({
        ...ScreenOptionsDictionary[Screens.CreatePIN],
        title: t('Screens.NameWallet'),
      }),
      component: NameWallet,
    },
    {
      name: Screens.UseBiometry,
      options: () => ({
        ...ScreenOptionsDictionary[Screens.CreatePIN],
        title: t('Screens.Biometry'),
      }),
      component: useBiometry,
    },
    {
      name: Screens.UsePushNotifications,
      options: () => ({
        ...ScreenOptionsDictionary[Screens.CreatePIN],
        title: t('Screens.UsePushNotifications'),
      }),
      component: PushNotification,
    },
    {
      name: Screens.Developer,
      component: Developer,
      options: () => {
        return {
          ...ScreenOptionsDictionary[Screens.Developer],
          title: t('Screens.Developer'),
          headerBackAccessibilityLabel: t('Global.Back'),
        }
      },
    },
  ]

  return (
    <Stack.Navigator initialRouteName={Screens.Splash} screenOptions={{ ...defaultStackOptions }}>
      {screens.map((item) => {
        return <Stack.Screen key={item.name} {...item} />
      })}
    </Stack.Navigator>
  )
}

export default OnboardingStack
