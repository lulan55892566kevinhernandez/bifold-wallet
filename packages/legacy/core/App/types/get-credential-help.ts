import { StackNavigationProp } from '@react-navigation/stack'

import { NotificationStackParams, Screens } from './navigators'

export type GetCredentialHelpAction = (
  navigation: StackNavigationProp<NotificationStackParams, Screens.ProofRequest>
) => void

export interface GetCredentialHelpEntry {
  credDefIds: string[]
  schemaIds: string[]
  action: GetCredentialHelpAction
}
