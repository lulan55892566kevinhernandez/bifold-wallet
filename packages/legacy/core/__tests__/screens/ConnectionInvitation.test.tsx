import { useConnections } from '@aries-framework/react-hooks'
import { useNavigation } from '@react-navigation/core'
import { render } from '@testing-library/react-native'
import React from 'react'

import { ConfigurationContext } from '../../App/contexts/configuration'
import ConnectionInvitation from '../../App/screens/ConnectionInvitation'
import { Screens } from '../../App/types/navigators'
import configurationContext from '../contexts/configuration'

jest.mock('@react-navigation/core', () => {
  return require('../../__mocks__/custom/@react-navigation/core')
})
jest.mock('@react-navigation/native', () => {
  return require('../../__mocks__/custom/@react-navigation/native')
})

describe('ConnectionInvitation screen', () => {
  beforeEach(() => {
    jest.clearAllMocks()
    // @ts-ignore
    useConnections.mockReturnValue({ records: [] })
  })

  test('Renders correctly', async () => {
    const tree = render(
      <ConfigurationContext.Provider value={configurationContext}>
        <ConnectionInvitation navigation={useNavigation()} route={{ key: '', name: Screens.ConnectionInvitation }} />
      </ConfigurationContext.Provider>
    )

    expect(tree).toMatchSnapshot()
  })
})
