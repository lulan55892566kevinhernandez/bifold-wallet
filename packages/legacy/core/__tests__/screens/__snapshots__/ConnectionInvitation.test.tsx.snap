// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionInvitation screen Renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
    ]
  }
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flexGrow": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginHorizontal": 30,
            "marginVertical": 20,
            "padding": 16,
            "textAlign": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontSize": 28,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Connection.ScanQR
        </Text>
        <Text
          style={
            Object {
              "color": "#000000",
              "fontSize": 20,
              "fontWeight": "normal",
              "marginTop": 8,
              "textAlign": "center",
            }
          }
        >
          Connection.ScanQRComment
        </Text>
      </View>
      <View
        style={
          Object {
            "flexGrow": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#42803E",
              "borderRadius": 40,
              "borderWidth": 10,
              "height": 730,
              "justifyContent": "center",
              "marginHorizontal": 10,
              "width": 730,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
            testID="com.ariesbifold:id/LoadingActivityIndicator"
          >
            <Image
              source={
                Object {
                  "default": "",
                }
              }
              style={
                Object {
                  "height": "33%",
                  "width": "33%",
                }
              }
              testID="com.ariesbifold:id/LoadingActivityIndicatorImage"
            />
            <View
              collapsable={false}
              nativeID="animatedComponent"
              style={
                Object {
                  "position": "absolute",
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <
                fill="#FFFFFF"
                height={200}
                width={200}
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
